src/daemon.py

```diff
@@ -199,6 +203,19 @@ def write_coingecko_prices(
         logger.error("Error inserting record: %s", e)
 
 
+def get_price(
+    PriceProviders: List, block_number: int, token_address: str
+) -> Optional[Tuple[float, str]]:
+    for provider in PriceProviders:
+        try:
+            price = provider.get_price(block_number, token_address)
+            if price is not None:
+                return price, provider.__class__.__name__
```

Comment:
Using those double underscore methods is a bit unconventional. If you need a name per provider, implementing a `provider.name` function would be more appropriate.

This could also be handled by using an abstract base class for providers which implements a `name` function, using your implementation from here. Making all providers a subclass, should prevent code duplication.